### DISCUSSION TITLE
image 전송과 함께 데이터는 json으로 보내는 API 를 분리

### DISCUSSION THREAD

**Discussion by june0216** [EFUB-LakkuLakku/LakkuLakku-Back#114] [Ideas]
본 프로젝트는 데이터 JSON과 이미지(form-data)를 하나의 API를 통해 전송합니다. 
하지만 "보통 이미지와 JSON데이터를 함께 보내지 않습니다. API를 전송할 때는 보통 content/type을 application/json으로 전송하는데 이것은 multipart/formdata 방식이 아니기 때문입니다." 이라고 합니다

이에 따라, 방법은 2가지 인데
1) 분리해서 요청하기 
서버는 이미지를 저장하고, 저장된 이미지의 id(또는 파일명)를 클라이언트에 전송합니다.
클라이언트는 이미지의 id와 전송할 데이터를 application/json으로 전송합니다.
2) base64 인코딩
이미지를 base64로 인코딩해서 application/json으로 전송할 데이터와 base64로 인코딩된 이미지를 함께 전송합니다.

이 중 하나를 골라서 개선해야할 것 같습니다!
찾아보니 2번 방법은 디비에 base64 로 저장할경우, 차지하는 용량이 많이 늘고, 트래픽도 는다는 단점이 있다고 합니다! 이에 따라 개인적인 의견으로는 1번이 보다 적합할 것 같습니다!

**Comment by nitronium102** [comment]
오 안 그래도 전에 이것 관련해서 오류 났었던 것 같은데 짚어주셔서 감사합니다!
관련해서 좀 더 찾아봤는데 저도 1번이 보다 적합할 것 같아요👍👍

**Reply by june0216** [reply]
칼답변 감사합니다~! 👍 다음 코드 리팩터링할 때 고쳐보면 좋을 것 같습니다~!
